packages/design-system/src/components/NativeDialog/NativeDialog.tsx

```diff
@@ -116,7 +116,7 @@ export const NativeDialog = ({
         rect.left <= event.clientX &&
         event.clientX <= rect.left + rect.width;
       if (!isInDialog) {
-        exit();
+        exit(event);
```

Comment:
backdropClickExits was throwing an error in the define method because the event.target did not exist. Hence passing in the event here solved the issue.